Data Structures/Linked Lists/Insert in specific position/Solution.java

```diff
@@ -0,0 +1,116 @@
+import java.io.*;
+import java.math.*;
+import java.security.*;
+import java.text.*;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.regex.*;
+
+public class Solution {
+
+    static class SinglyLinkedListNode {
+        public int data;
+        public SinglyLinkedListNode next;
+
+        public SinglyLinkedListNode(int nodeData) {
+            this.data = nodeData;
+            this.next = null;
+        }
+    }
+
+    static class SinglyLinkedList {
+        public SinglyLinkedListNode head;
+        public SinglyLinkedListNode tail;
+
+        public SinglyLinkedList() {
+            this.head = null;
+            this.tail = null;
+        }
+
+        public void insertNode(int nodeData) {
+            SinglyLinkedListNode node = new SinglyLinkedListNode(nodeData);
+
+            if (this.head == null) {
+                this.head = node;
+            } else {
+                this.tail.next = node;
+            }
+
+            this.tail = node;
+        }
+    }
+
+    public static void printSinglyLinkedList(SinglyLinkedListNode node, String sep, BufferedWriter bufferedWriter) throws IOException {
+        while (node != null) {
+            bufferedWriter.write(String.valueOf(node.data));
+
+            node = node.next;
+
+            if (node != null) {
+                bufferedWriter.write(sep);
+            }
+        }
+    }
+
+    // Complete the insertNodeAtPosition function below.
+
+    /*
+     * For your reference:
+     *
+     * SinglyLinkedListNode {
+     *     int data;
+     *     SinglyLinkedListNode next;
+     * }
+     *
+     */
+    static SinglyLinkedListNode insertNodeAtPosition(SinglyLinkedListNode head, int data, int position) {
+        SinglyLinkedListNode auxPrevious = head;
+        SinglyLinkedListNode auxNext = head.next;
+        int cont = 0;
+        while (cont != position - 1) {
+            auxPrevious = auxPrevious.next;
+            auxNext = auxPrevious.next;
+            cont += 1;
+        }
+        if (cont <= 1000) {
+            SinglyLinkedListNode newNext = auxNext;
+            SinglyLinkedListNode newNode = new SinglyLinkedListNode(data);
+            auxPrevious.next = newNode;
+            newNode.next = newNext;
+        }
+        return head;
```

Comment:
Add relevant line spaces in between the lines to make it better. Here's an example:


```
SinglyLinkedListNode auxPrevious = head;
SinglyLinkedListNode auxNext = head.next;
// NOTICE THIS LINE SPACE        
int cont = 0;
```